app/models/manageiq/providers/openstack/infra_manager/refresh_parser.rb

```diff
@@ -96,7 +96,10 @@ def all_stack_server_resources(stack)
         # TODO(lsmola) loading this from already obtained nested stack hierarchy will be more effective. This is one
         # extra API call. But we will need to change order of loading, so we have all resources first.
         # Nested depth 50 just for sure, although nobody should nest templates that much
-        @orchestration_service.list_resources(:stack => stack, :nested_depth => 50).body['resources']
+        # To further speed up the query we only search for those resources we care about - those whose
+        # physical_resource_id matches the id of a nova server
+        server_ids = servers.map{|s| s.id}
+        @orchestration_service.list_resources(:stack => stack, :nested_depth => 50, :physical_resource_id => server_ids).body['resources']
```

Comment:
@Ladas it does, but there's no speed improvement, whereas this filter reduced the time of the query considerably